src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -4997,7 +4997,10 @@ namespace SurfaceGeometry {
                                        cCurrentModuleObject,
                                        state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Name,
                                        SurfaceNumProp));
+                // For the remaining of the routine, as soon as ErrorsFound is true, we just skip to the next subsurface
+                // no point in more checks, the subsurface is invalid, and it could lead to a crash (cf #9331). Eventually, a Fatal will be issued
                 ErrorsFound = true;
+                continue;
```

Comment:
I don't have a problem with using a continue when "very" severe errors are found. But I wonder if there are some errors that won't show up until the user fixes the first one found. It may not be necessary to continue on some of these warnings so that as many warnings could be reported the first time through. Which one of these input errors actually causes the crash?

Reply:
I understand the sentiment here.  If we could issue all 3 errors up front on the subsurface, it would be good to let the user know all 3 errors right away.  Otherwise they will end up possibly running 3 iterations to solve all the issues.  But I also understand the idea of wanting to avoid issues from subsequent checking.  I could be swayed either way.  Perhaps some extra description of the exact errors we are bypassing by each `continue` would help.

Reply:
I am perfectly fine if we revert https://github.com/NREL/EnergyPlus/pull/9646/commits/dcbda05e987d9151667f8aa8348b4d939a5618c3

Reply:
Side note: I actually have a stashed commit that went even further and wrapped ever GetXXX call inside and `if (!ErrorsFound)` so it'll bail early here: https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/SurfaceGeometry.cc#L1284-L1355

Reply:
@jmarrec I think I will take the offer of just reverting the commit.  That will allow the fix to go in without having to worry about this new pattern.  I'll revert and test it out locally to make sure, then push it up and get this merged.